src/frontend/js/clusive-prefs-django-store.js

```diff
@@ -0,0 +1,167 @@
+        gradeNames: ['fluid.dataSource'],
+        storeConfig: {
+            getURL: '/account/prefs',
+            setURL: '/account/prefs',
+            resetURL: '/account/prefs/profile'
+        },
+        components: {
+            encoding: {
+                type: 'fluid.dataSource.encoding.none'
+            },            
+            messageQueue: {
+                type: "clusive.djangoMessageQueue"
+            }
+            
+        },
+        listeners: {
+            'onRead.impl': {
+                listener: 'clusive.prefs.djangoStore.getUserPreferences',
+                args: ['{arguments}.1', "{that}.messageQueue"]
+            }
+        },
+        invokers: {
+            get: {
+                args: ['{that}', '{arguments}.0', '{that}.options.storeConfig']
+            }
+        }
+    });
+
+    fluid.defaults('clusive.prefs.djangoStore.writable', {
+        gradeNames: ['fluid.dataSource.writable'],
+        listeners: {
+            'onWrite.impl': {
+                listener: 'clusive.prefs.djangoStore.setUserPreferences',
+                args: ["{arguments}.0", "{arguments}.1", "{that}.messageQueue"]
+            }
+        },
+        invokers: {
+            set: {
+                args: ['{that}', '{arguments}.0', '{arguments}.1', '{that}.options.storeConfig']
+            }
+        }
+    });
+
+    fluid.makeGradeLinkage('clusive.prefs.djangoStore.linkage', ['fluid.dataSource.writable', 'clusive.prefs.djangoStore'], 'clusive.prefs.djangoStore.writable');
+
+    clusive.prefs.djangoStore.getUserPreferences = function(directModel, messageQueue) {
+        console.debug('clusive.prefs.djangoStore.getUserPreferences', directModel, messageQueue);
+
+        var djangoStorePromise = fluid.promise();
+
+        var preferenceChangeMessages = [].concat(messageQueue.queue).filter(function (item) {
+            if(item.content.type === "PC") {
+                return true;
+            }
+        }) 
+        
+        if(preferenceChangeMessages.length > 0) {
+            // Check for any preference changes in the message queue; 
+            // if there are any, the latest is them, not what's 
+            // available from the server; return the last preference change 
+            // from the queue
+            var latestPreferences = preferenceChangeMessages.pop().content.preferences;
+            console.debug("Got user preferences from the message queue", latestPreferences);
+            djangoStorePromise.resolve({
+                preferences: latestPreferences
+            });            
+            
+        } else {
+            var getURL = directModel.getURL;
+        
+            $.get(getURL, function(data) {
+                console.debug('Get user preferences from the server ', getURL);
+                console.debug('Received preferences: ', data);
+                djangoStorePromise.resolve({
+                    preferences: data
+                });
```

Comment:
Looks like this doesn't do any caching of response, so will not resolve the issue of multiple server calls to get preferences. That will be a separate fix later?

Reply:
I'll add a debounce function here.